### PR TITLE
Use travis to publish to bintray

### PR DESCRIPTION
**Pull request by davidillsley**
Taking inspiration from [1] and [2], this PR automates publishing to bintray from travis-ci. It works by publishing builds triggered by tags, allowing for the current jenkins bits to continue to work.

In order to trigger a release, push a tag eg pushing [3] resulted in [4], and [5]

It currently publishes a pretty sparse POM, and I'm sure there are lots of possible improvements, but this seemed like a reasonable start.

It's using my bintray creds to publish to the alphagov org on bintray. Shout if you want me to add a few more peoples bintray accounts to that org.

[1] http://bithacker.org/post/120996678805/publishing-artifacts-to-bintray-from-travis-ci
[2] https://github.com/bintray/bintray-examples/blob/master/gradle-bintray-plugin-examples/publications-example/build.gradle
[3] https://github.com/davidillsley/dropwizard-logstash/tree/1.0.0
[4] https://travis-ci.org/davidillsley/dropwizard-logstash/builds/210472908
[5] https://bintray.com/alphagov/maven-test/dropwizard-logstash/1.0.5-1.0.0

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 language: java
-
 jdk:
-  - oraclejdk8
+- oraclejdk8
+after_success:
+- test "${TRAVIS_PULL_REQUEST}" == "false" && test "${TRAVIS_TAG}" != "" && BUILD_NUMBER=$TRAVIS_TAG
```
Comment (rhowe-gds) [inline .travis.yml]: Can simplify into this, I think
`test "$TRAVIS_PULL_REQUEST" == false && test -n "$TRAVIS_TAG" && BUILD_NUMBER=$TRAVIS_TAG`

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Fri Feb 28 16:53:33 GMT 2014
+#Sun Mar 12 23:33:03 GMT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-1.11-bin.zip
+distributionUrl=http\://services.gradle.org/distributions/gradle-3.4.1-bin.zip
```
Comment (rhowe-gds) [inline gradle/wrapper/gradle-wrapper.properties]: No HTTPS? :(

File: build.gradle [hunk 3]
```diff
@@ -33,14 +36,29 @@ publishing {
...
+        maven {
+            url "file:/srv/maven" // change to point to your repo, e.g. http://my.org/repo
         }
+    }
+}
 
+bintray {
+	user = System.getenv('BINTRAY_USER')
+	key = System.getenv('BINTRAY_API_KEY')
+  publications = ['mavenJava']
```
Comment (rhowe-gds) [inline build.gradle]: Whitespace nitpick?

File: .travis.yml (hunk 1 above)
Comment (davidillsley) [inline .travis.yml]: Good call.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (davidillsley) [inline gradle/wrapper/gradle-wrapper.properties]: :( that's what gradlew wrapper put in there. Will fix

File: build.gradle (hunk 3 above)
Comment (davidillsley) [inline build.gradle]: Sure. Updates inbound.

**Comment by willp-bl** [comment]
Looks good to me 👍

**Comment by rhowe-gds** [comment]
Looks good to me as well

**Comment by philandstuff** [comment]
@davidillsley I'd like you to add a few more accounts to that bintray org 😄

**Comment by philandstuff** [comment]
I admit this shout is a bit later than ideal...

**Pull request merged**